--- icu/source/tools/toolutil/pkg_genc.c.orig	2009-10-05 20:42:46.878010257 +0200
+++ icu/source/tools/toolutil/pkg_genc.c	2009-10-05 20:43:44.073164845 +0200
@@ -118,10 +118,18 @@ static const struct AssemblyType {
 } assemblyHeader[] = {
     {"gcc",
         ".globl %s\n"
+#ifdef __arm__
+        "\t.section .note.GNU-stack,\"\",%%progbits\n"
+#else
         "\t.section .note.GNU-stack,\"\",@progbits\n"
+#endif
         "\t.section .rodata\n"
         "\t.align 8\n" /* Either align 8 bytes or 2^8 (256) bytes. 8 bytes is needed. */
+#ifdef __arm__
+        "\t.type %s,%%object\n"
+#else
         "\t.type %s,@object\n"
+#endif
         "%s:\n\n",
 
         ".long ","",HEX_0X
